eclair-core/src/main/scala/fr/acinq/eclair/db/migration/CompareDb.scala

```diff
@@ -0,0 +1,80 @@
+package fr.acinq.eclair.db.migration
+
+import fr.acinq.eclair.db.DualDatabases
+import fr.acinq.eclair.db.pg.PgUtils
+import grizzled.slf4j.Logging
+import scodec.bits.ByteVector
+
+import java.sql.{Connection, ResultSet}
+
+object CompareDb extends Logging {
+
+  def compareTable(conn1: Connection,
+                   conn2: Connection,
+                   table1: String,
+                   table2: String,
+                   hash1: ResultSet => ByteVector,
+                   hash2: ResultSet => ByteVector): Boolean = {
+    val rs1 = conn1.prepareStatement(s"SELECT * FROM $table1").executeQuery()
+    val rs2 = conn2.prepareStatement(s"SELECT * FROM $table2").executeQuery()
+
+    var hashes1 = List.empty[ByteVector]
+    while (rs1.next()) {
+      hashes1 = hash1(rs1) +: hashes1
+    }
+
+    var hashes2 = List.empty[ByteVector]
+    while (rs2.next()) {
+      hashes2 = hash2(rs2) +: hashes2
+    }
+
+    val res = hashes1.sorted == hashes2.sorted
+    val diff1 = hashes1 diff hashes2
+    val diff2 = hashes2 diff hashes1
```

Comment:
nit: you can put that in the `else` branch below, no need to compare a `diff` on these big lists if everything if ok?

Reply:
Done with 5c3eac546.